src/app/clusters/time-format-localization-server/time-format-localization-logic.cpp

```diff
@@ -0,0 +1,222 @@
+
+    attrPersistence.LoadNativeEndianValue<HourFormatEnum>(
+        { kRootEndpointId, TimeFormatLocalization::Id, TimeFormatLocalization::Attributes::HourFormat::Id }, mHourFormat,
+        kDefaultHourFormat);
+}
+
+CHIP_ERROR TimeFormatLocalizationLogic::GetSupportedCalendarTypes(AttributeValueEncoder & aEncoder) const
+{
+    DeviceLayer::DeviceInfoProvider * provider = DeviceLayer::GetDeviceInfoProvider();
+    VerifyOrReturnValue(provider != nullptr, aEncoder.EncodeEmptyList());
+
+    AutoReleaseIterator it(provider->IterateSupportedCalendarTypes());
+    VerifyOrReturnValue(it.IsValid(), aEncoder.EncodeEmptyList());
+
+    return aEncoder.EncodeList([&it](const auto & encoder) -> CHIP_ERROR {
+        CalendarTypeEnum type;
+
+        while (it.Next(type))
+        {
+            ReturnErrorOnFailure(encoder.Encode(type));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+bool TimeFormatLocalizationLogic::IsSupportedCalendarType(CalendarTypeEnum reqCalendar, CalendarTypeEnum * validCalendar)
+{
+    DeviceLayer::DeviceInfoProvider * provider = DeviceLayer::GetDeviceInfoProvider();
+    VerifyOrReturnValue(provider != nullptr, false);
+
+    AutoReleaseIterator it(provider->IterateSupportedCalendarTypes());
+    VerifyOrReturnValue(it.IsValid(), false);
+
+    CalendarTypeEnum type;
+    bool found = false;
+
+    while (it.Next(type))
+    {
+        // Update the optional validCalendar to a value from the SupportedList.
+        // This will return either the last element of the list or the requested
+        // value if exists.
+        if (validCalendar != nullptr)
+        {
+            *validCalendar = type;
+        }
+
+        if (type == reqCalendar)
+        {
+            found = true;
+            break;
+        }
+    }
+
+    return found;
+}
+
+TimeFormatLocalization::CalendarTypeEnum TimeFormatLocalizationLogic::GetActiveCalendarType() const
+{
+    return mCalendarType;
+}
+
+DataModel::ActionReturnStatus TimeFormatLocalizationLogic::WriteAttribute(const DataModel::WriteAttributeRequest & request,
+                                                                          AttributePersistenceProvider & attrProvider,
+                                                                          AttributeValueDecoder & decoder)
+{
+    AttributePersistence attrPersistence{ attrProvider };
+
+    switch (request.path.mAttributeId)
+    {
+    case TimeFormatLocalization::Attributes::HourFormat::Id: {
+        CHIP_ERROR result = attrPersistence.DecodeAndStoreNativeEndianValue(
+            { kRootEndpointId, TimeFormatLocalization::Id, HourFormat::Id }, decoder, mHourFormat);
+        return result == CHIP_NO_ERROR ? Protocols::InteractionModel::Status::Success
+                                       : Protocols::InteractionModel::Status::ConstraintError;
+    }
+
+    case TimeFormatLocalization::Attributes::ActiveCalendarType::Id: {
+        if (!mFeatures.Has(TimeFormatLocalization::Feature::kCalendarFormat))
```

Comment:
API requirement is that writes are done only on items return by ::Attributes. This check is not needed.

Reply:
Removing check.